Doc/library/py_compile.rst

```diff
@@ -65,6 +68,36 @@ byte-code cache files in the directory containing the source code.
       caveat that :exc:`FileExistsError` is raised if *cfile* is a symlink or
       non-regular file.
 
+   .. versionchanged:: 3.7
+      The *invalidation_mode* parameter was added as specified in :pep:`552`.
+
+
+.. class:: PycInvalidationMode
+
+   A enumeration of possible methods the interpreter can use to determine
+   whether a bytecode file is up to date with a source file. The pyc indicates
+   the desired invalidation mode in its header. See :ref:`pyc-invalidation` for
+   more information on how Python invalidates pycs at runtime.
+
+   .. versionadded:: 3.7
+
+   .. attribute:: TIMESTAMP
+
+      The pyc should include the timestamp and size of the source file, which
+      Python will compare against the metadata of the source file at runtime to
+      determine if the pyc needs to be regenerated.
+
+   .. attribute:: CHECKED_HASH
+
+      The pyc should include a hash of the source file which Python will compare
+      against the source at runtime to determine if the pyc needs to be
+      regenerated.
+
+   .. attribute:: UNCHECKED_HASH
```

Comment:
> The text below merely says "a hash is included but not validated" but makes no mention of what is validated in this situation. the pyc-invalidation docs cover this in more detail. It seems worth repeating that this just means a hash is generated when generating a pyc but is not checked by default here.

Yeah, I was trying to funnel everything into `import.rst`. I will beef up this part a big, though.

> It'd also be good to include a statement of when this might be useful (packaging distribution installed pre-generated pyc files where something else is guaranteeing they cannot go out of sync for example?). Though perhaps that description belongs in import.rst's pyc-invalidation section.

I put some commentary about why you might want this in the what's new? entry. Maybe, it should go somewhere more "permanent", too, though.